### PR TITLE
CT-220 - skeleton for algorand

### PR DESCRIPTION
**Pull request by twtaylorbitgo**
Intended as a skeleton for Algorand work. This includes algosdk, which will be used for transaction signing.

### CODE REVIEW

File: src/v2/coins/algo.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+
+class Algorand extends BaseCoin {
+
+    constructor() {
+      super();
+    }
+
+    getChain() {
+        return "algo";
```
Comment (CapnMigraine) [inline src/v2/coins/algo.ts]: nit: use single quotes here and down in ln:16

File: src/v2/coins/algo.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+     * Flag for sending value of 0
+     * @returns {boolean} True if okay to send 0 value, false otherwise
+     */
+    valuelessTransferAllowed(): boolean {
+        // TODO: this sounds like its true with the staking txes - confirm before launch
+        return false;
+    }
+}
+
+export default Algorand;
```
Comment (CapnMigraine) [inline src/v2/coins/algo.ts]: nit: add new line

File: src/v2/coins/talgo.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    constructor() {
+      super();
+    }
+
+    getChain() {
+        return 'talgo';
+      }
+}
+
+module.exports = TAlgorand;
```
Comment (CapnMigraine) [inline src/v2/coins/talgo.ts]: nit: add new line

File: src/v2/coins/algo.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
+import BaseCoin = require('../baseCoin');
+import * as _ from 'lodash';
+
+
+class Algorand extends BaseCoin {
```
Comment (CapnMigraine) [inline src/v2/coins/algo.ts]: Can we use `Algo` as class name for consistency?

File: src/v2/coins/talgo.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import Algorand from './algo';
+
+class TAlgorand extends Algorand {
```
Comment (CapnMigraine) [inline src/v2/coins/talgo.ts]: Let's use `Talgo` for consistency

File: src/v2/coins/talgo.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+    constructor() {
+      super();
+    }
+
+    getChain() {
+        return 'talgo';
+      }
+}
```
Comment (CapnMigraine) [inline src/v2/coins/talgo.ts]: Please also add:
```
  getFullName() {
    return 'Testnet Algorand';
  }
```

Comment (twtaylorbitgo) [reply src/v2/coins/talgo.ts]: ack

File: src/v2/coins/algo.ts (hunk 4 above)
Comment (twtaylorbitgo) [reply src/v2/coins/algo.ts]: sure

File: src/v2/coins/talgo.ts (hunk 3 above)
Comment (twtaylorbitgo) [reply src/v2/coins/talgo.ts]: ack

File: src/v2/coins/algo.ts (hunk 1 above)
Comment (twtaylorbitgo) [reply src/v2/coins/algo.ts]: ack

File: src/v2/coins/algo.ts (hunk 2 above)
Comment (twtaylorbitgo) [reply src/v2/coins/algo.ts]: ack

**Pull request merged**